completion/api/v1/views.py

```diff
@@ -33,6 +34,7 @@ class CompletionBatchView(APIView):
     """
     Handles API requests to submit batch completions.
     """
+    authentication_classes = (JwtAuthentication,)
```

Comment:
Do we also need to add the existing method of auth now that we are using authentication_classes?

Reply:
Yes, if you want to continue to support session authentication you will need to add that to the `authentication_classes` tuple. What are the known clients of this API?

Reply:
I am not sure about all the known clients so far I think it is being used in mobile and probably web.

Reply:
Okay, it's probably a good idea to be sure of the existing clients, so you can be sure which authentication mechanisms you need to support.

Reply:
The mobile app should be the first client of this batch completion endpoint, nothing else uses it as far as I know.  But the front-end/web of edx-platform IS using the SubsectionCompletionView below, so you'll have to add session auth to the `authentication_classes` there, too.

Reply:
If you want mobile apps to access this, you'll need to add `OAuth2AuthenticationAllowInactiveUser`.

Reply:
@nasthagiri as per my knowledge `OAuth2AuthenticationAllowInactiveUser` is part of edx-platform. Are you suggesting that we should duplicate the code for that class here and use that as an authentication class as well?